### PR TITLE
chore(style): Adds travis driven awesome-lint

### LINKED ISSUE #18: Use awesome-lint?
https://github.com/sindresorhus/awesome-lint

:)

### PR DESCRIPTION
**Pull request by joshwiens**
- Adds `awesome-lint`
- Adds editorconfig
- Adds Travis build executing `awesome-lint` in npm test
- Sets block ignore for `list-item-indent` on the t.o.c
- Fixes style errors
- Enables branch protection for `master` setting travis build to `required`

Resolves #18

### CODE REVIEW

File: .remarkrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "plugins": {
+    "remark-lint": {
+      "list-item-indent": false
```
Comment (kentcdodds) [inline .remarkrc]: With this, do you still need the inline comment in the README anymore?

Comment (joshwiens) [inline .remarkrc]: Doesn't read the rc file, going to have to use the inline ignores

**Comment by joshwiens** [comment]
Don't land this yet, need to fix the package.json `contributors` field

File: package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "type": "git",
+    "url": "git+https://github.com/d3viant0ne/awesome-webpack.git"
+  },
+  "keywords": [
+    "awesome",
+    "webpack"
+  ],
+  "author": "Joshua Wiens <d3viant0ne>",
+  "contributors": [
+  {
```
Comment (kentcdodds) [inline package.json]: Preeety sure we don't want this to be nested in an object.

Comment (kentcdodds) [inline package.json]: > Don't land this yet, need to fix the package.json contributors field

Oh, you knew that 😅

**Comment by kentcdodds** [comment]
LGTM

**Pull request merged**